plugins/kubernetes/test/models/kubernetes/template_filler_test.rb

```diff
@@ -786,6 +786,16 @@ def secret_annotations(hash)
       end
     end
 
+    describe "podtemplate" do
+      before do
+        raw_template.replace(YAML.safe_load(read_kubernetes_sample_file('kubernetes_podtemplate.yml')).deep_symbolize_keys)
+      end
+
+      it "works" do
+        template.to_hash
```

Comment:
test that the container is modified in some way, this test would have passed before adding the new code
also test for replicas not being set

Reply:
Also added in test that `:spec` is not filled, given `:template` is the top level element